### PR TITLE
Task/gps logic

### PR DESCRIPTION
**Pull request by b-goldsmith**
Built out receiveGPS. The BusAlertService.java has the main business logic for this task.

### CODE REVIEW

**Comment by b-goldsmith** [comment]
I resolved conflicts. This should be good to merge with main.

**Comment by b-goldsmith** [comment]
tested locally and this looks good. I'm going to try loading it in heroku and doing a quick test

**Comment by b-goldsmith** [comment]
okay, this has been tested locally and on heroku and is working properly.

File: src/main/java/server/BusAlert/BusAlertService.java [hunk 1]
```diff
@@ -1,10 +1,142 @@
...
+                            .getRoute()
+                            .getStops()
+                            .get(stop.getRoute().getStops().indexOf(stop) + 2)
+                            .getRiders()
+                            .parallelStream()
+                            .map(rider ->
+                                    riderService
+                                            .notifyRider(rider)
+                            )
+                            .filter(Objects::nonNull)
```
Comment (garrett-vangilder) [inline src/main/java/server/BusAlert/BusAlertService.java:61]: this is slick, but `notififyRider` already prints an error message, wouldn't running `failedRiderNotifications` just duplicate our logs?

File: src/main/java/server/BusAlert/BusAlertService.java [hunk 2]
```diff
@@ -1,10 +1,142 @@
...
+                ? env_unit.charAt(0) : 'K';
+
+        // limit is used to determine the farther from a bus stop
+        // a bus can be and still be considered at the stop
+        // Pull ENVIRONMENTAL VAR
+        String env_limit = System.getenv("LIMIT");
+        // If ENVIRONMENTAL VAR is valid use that, otherwise use 0.25 kilometeres
+        double limit = env_limit != null ? Double.parseDouble(env_limit) : 0.25;
+        double theta = lon1 - lon2;
+        double dist = Math.sin(deg2rad(lat1)) * Math.sin(deg2rad(lat2)) + Math.cos(deg2rad(lat1)) * Math.cos(deg2rad(lat2)) * Math.cos(deg2rad(theta));
```
Comment (garrett-vangilder) [inline src/main/java/server/BusAlert/BusAlertService.java:125]: oh boy

**Review by garrett-vangilder — APPROVED** [review]

**Pull request merged**